### PR TITLE
Convert view internals to TS

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: packages/@ember/-internals/glimmer/lib/component.ts [hunk 1]
```diff
@@ -857,7 +857,7 @@ class Component
       if (!lazyEventsProcessedForComponentClass.has(proto)) {
         let lazyEvents = eventDispatcher.lazyEvents;
 
-        lazyEvents.forEach((mappedEventName: string, event: string) => {
+        lazyEvents.forEach((mappedEventName: string | null, event: string) => {
```
Comment (wagenet) [inline packages/@ember/-internals/glimmer/lib/component.ts:860]: Given the check on the next line we definitely expect a possible null here.

File: packages/@ember/-internals/views/lib/mixins/action_support.ts [hunk 2]
```diff
@@ -5,8 +5,16 @@ import { inspect } from '@ember/-internals/utils';
...
 
-const mixinObj = {
-  send(actionName, ...args) {
+/**
+ @class ActionSupport
+ @namespace Ember
+ @private
+*/
+interface ActionSupport {
+  send(actionName: string, ...args: unknown[]): void;
```
Comment (wagenet) [inline packages/@ember/-internals/views/lib/mixins/action_support.ts:14]: @chriskrycho's favorite!

File: packages/@ember/-internals/views/lib/system/event_dispatcher.ts [hunk 3]
```diff
@@ -239,50 +249,58 @@ export default EmberObject.extend({
...
+        if (actionState) {
+          actions = [actionState];
+        }
       }
 
       // We have to check for actions here since in some cases, jQuery will trigger
       // an event on `removeChild` (i.e. focusout) after we've already torn down the
       // action handlers for the view.
       if (!actions) {
+        // FIXME: This branch is never called in tests. Improve tests or remove
```
Comment (wagenet) [inline packages/@ember/-internals/views/lib/system/event_dispatcher.ts:303]: Not sure if there's anything more we should do about this right now.

File: packages/@ember/-internals/glimmer/lib/component.ts (hunk 1 above)
Comment (chriskrycho) [reply packages/@ember/-internals/glimmer/lib/component.ts:860]: Out of curiosity, does `lazyEvents` itself not have the right type? Seems like we want it to be there instead?

File: packages/@ember/-internals/views/lib/mixins/action_support.ts (hunk 2 above)
Comment (chriskrycho) [reply packages/@ember/-internals/views/lib/mixins/action_support.ts:14]: ***BURN IT TO THE GROUND. SALT THE GROUND WHERE IT STOOD. GLASS THE GROUND FROM ORBIT.***

File: packages/@ember/-internals/views/lib/mixins/view_support.ts [hunk 4]
```diff
@@ -3,12 +3,34 @@ import { descriptorForProperty, Mixin, nativeDescDecorator } from '@ember/-inter
 import { assert } from '@ember/debug';
 import { hasDOM } from '@ember/-internals/browser-environment';
 import { matches } from '../system/utils';
+import type Component from '@ember/component';
+import { View } from '@ember/-internals/glimmer/lib/renderer';
+import { SimpleElement } from '@simple-dom/interface';
 
-function K() {
+function K(this: unknown) {
```
Comment (chriskrycho) [inline packages/@ember/-internals/views/lib/mixins/view_support.ts:10]: BURN IT. GAHHHHH.

File: packages/@ember/-internals/views/lib/system/event_dispatcher.ts (hunk 3 above)
Comment (chriskrycho) [reply packages/@ember/-internals/views/lib/system/event_dispatcher.ts:303]: Huh. Is it even an issue anymore given that we don't have jQuery *integration* anymore?

**Review by chriskrycho — APPROVED** [review]

File: packages/@ember/-internals/glimmer/lib/component.ts (hunk 1 above)
Comment (wagenet) [reply packages/@ember/-internals/glimmer/lib/component.ts:860]: Types are correct. Looks like this ends up being redundant.

File: packages/@ember/-internals/views/lib/system/event_dispatcher.ts (hunk 3 above)
Comment (wagenet) [reply packages/@ember/-internals/views/lib/system/event_dispatcher.ts:303]: I suspect we just don't need it. But am also a bit hesitant to remove it without fully understanding.

**Pull request merged**